networkx/algorithms/connectivity/connectivity.py

```diff
@@ -325,8 +325,9 @@ def neighbors(v):
     kwargs = dict(flow_func=flow_func, auxiliary=H, residual=R)
 
     # Pick a node with minimum degree
-    degree = G.degree()
+    degree = dict(G.degree())
     minimum_degree = min(degree.values())
+    # TODO
```

Comment:
What about this:

```python
# Pick a node with minimum degree
v = min(G, key=G.degree)
# Node connectivity is bounded by degree.
K = G.degree(v)
``` 
I expect to be slightly slower, but in this function it represents a very small part of the total running time.